src/model/document.ts

```diff
@@ -88,7 +88,7 @@ export class Document implements Feature {
       this.identifiers.add(this.url);
     }
     this.kinds.add(`${this.parsedDocument.type}-document`);
-    // this._resolve();
+    this.warnings = base.warnings.slice();
```

Comment:
No, a ScannedDocument should be logically immutable by the time that we're creating Documents